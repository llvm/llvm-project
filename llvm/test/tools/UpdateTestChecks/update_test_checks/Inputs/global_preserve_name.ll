; RUN: opt -S < %s | FileCheck %s

@G = constant i32 42

;.
; CHECK: @G = constant i32 42
;.
define ptr @foo() {
; CHECK-LABEL: @foo(
; CHECK-NEXT:    ret ptr @G
;
  ret ptr @G
}
