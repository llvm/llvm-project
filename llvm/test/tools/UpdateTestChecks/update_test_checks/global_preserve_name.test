## Basic test checking that we capture existing lines matching global variable names
# RUN: cp -f %S/Inputs/global_preserve_name.ll %t.ll && %update_test_checks %t.ll --check-globals --preserve-names
# RUN: diff -u %t.ll %S/Inputs/global_preserve_name.ll.expected
## Verify that running without the --global-value-regex flag respects UTC_ARGS, and that the output is a fixed point.
# RUN: %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/global_preserve_name.ll.expected

