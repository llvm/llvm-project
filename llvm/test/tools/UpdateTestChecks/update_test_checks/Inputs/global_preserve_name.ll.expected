; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: -p --check-globals
; RUN: opt -S < %s | FileCheck %s

@G = constant i32 42

;.
; CHECK: @G = constant i32 42
;.
define ptr @foo() {
; CHECK-LABEL: @foo(
; CHECK-NEXT:    ret ptr @G
;
  ret ptr @G
}
